### PR TITLE
Task/03 components

### PR DESCRIPTION
**Pull request by 3apy6a**
- [x]  Mentor is added to pr
- [x]  AC copied to pr description
- [x]  Link to deployed version is added to pr
- [x]  The ProductsList component correctly receives product data as props.
- [x]  ProductCard component created and renders a single product.
- [x]  Header and Footer elements are present.
- [x]  ProductList displays right on mobile devices
[лучшій сайт](https://3apy6a.github.io/fe-react-2024/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,4 +1,7 @@
+import React from 'react';
```
Comment (vadym-zinchenko-moc) [inline src/App.tsx]: you can use `import { useState } from 'react';`

and just use `useState('Products')` istead `React.useState('Products')`

File: src/App.tsx [hunk 2]
```diff
@@ -7,14 +10,21 @@ import { ImageComponent } from './components/image/Image.component.tsx';
 import './App.module.css';
 
 function App() {
+    const [selectedNawBarItem, setSelectedNawBarItem] = React.useState('Products');
```
Comment (vadym-zinchenko-moc) [inline src/App.tsx]: you can use `import { useState } from 'react';`

and just use `useState('Products')` istead `React.useState('Products')`

File: src/App.tsx [hunk 3]
```diff
@@ -7,14 +10,21 @@ import { ImageComponent } from './components/image/Image.component.tsx';
 import './App.module.css';
 
 function App() {
```
Comment (vadym-zinchenko-moc) [inline src/App.tsx:12]: Screenshot 2024-05-08 at 18 44 04

Screenshot 2024-05-08 at 18 44 10

I see that your footer in one line with  content, maybe better do the same with header

File: src/App.tsx [hunk 4]
```diff
@@ -7,14 +10,21 @@ import { ImageComponent } from './components/image/Image.component.tsx';
...
+                <HeaderComponent setSelectedNawBarItem={setSelectedNawBarItem} />
                 <section>
-                    <div className="wrapper">
-                        <AboutComponent customClasses="w-6 w-12 offset-right-xs offset-mobile-none" />
-                        <ImageComponent customClasses="w-6 w-12 padding-120 padding-mobile-none" />
+                    <div className={'contentWrapper %{wrapper}'}>
+                        {selectedNawBarItem === 'About' ? (
+                            <>
+                                <AboutComponent customClasses="w-6 w-12 offset-right-xs offset-mobile-none" />
+                                <ImageComponent customClasses="w-6 w-12 padding-120 padding-mobile-none" />
```
Comment (vadym-zinchenko-moc) [inline src/App.tsx:23]: Screenshot 2024-05-08 at 18 04 30

something wrong with image position on desktop screen

File: src/components/header/components/desktop/Desktop.component.tsx [hunk 5]
```diff
@@ -7,67 +7,58 @@ import { LightIconComponent } from '@/components/header/components/shared/icons/
...
+                    <button className={styles.theme}>
+                        <DarkIconComponent customClasses={styles.offset_sm} />
+                    </button>
+                </li>
+            </ul>
+        </div>
+        <div className={`${styles.flex} ${styles.align_center}`}>
+            <ul className={styles.list}>
+                <li>
+                    <button className={`${styles.text_decoration} ${styles.color_white}`} onClick={() => setSelectedNawBarItem('About')}>
```
Comment (vadym-zinchenko-moc) [inline src/components/header/components/desktop/Desktop.component.tsx:37]: Screenshot 2024-05-08 at 18 04 43

you need to remove border when hovering over the button

File: src/components/image/Image.component.tsx [hunk 6]
```diff
@@ -1,6 +1,6 @@
 import React from 'react';
 
-import portrait from '@/assets/portrait.png';
+import portrait from '@/assets/portrait.svg';
```
Comment (vadym-zinchenko-moc) [inline src/components/image/Image.component.tsx:3]: what the reason to create Image component. You can just import image from assets into component and use, without extra components.

File: src/components/product/ProductCard.component.tsx [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+        <div>
+            <img className={styles.cardImg} src={product.images[0]} alt={'card'} />
+            <h2 className={styles.cardTitle}>{product.title}</h2>
+        </div>
+        <div className={styles.cardPriceWrapper}>
+            <p className={styles.cardPrice}>
+                {product.price}
+                <span className={styles.cardPricePrefix}>&#8372;</span>
+            </p>
+            <button className={styles.cardButton}>
```
Comment (vadym-zinchenko-moc) [inline src/components/product/ProductCard.component.tsx:17]: Screenshot 2024-05-08 at 18 40 46

you need to remove border when hovering over the button

**Review by vadym-zinchenko-moc — CHANGES_REQUESTED** [review]

File: src/App.tsx (hunk 1 above)
Comment (3apy6a) [reply src/App.tsx]: done

File: src/App.tsx (hunk 2 above)
Comment (3apy6a) [reply src/App.tsx]: done

File: src/components/header/components/desktop/Desktop.component.tsx (hunk 5 above)
Comment (3apy6a) [reply src/components/header/components/desktop/Desktop.component.tsx:37]: done

File: src/components/product/ProductCard.component.tsx (hunk 7 above)
Comment (3apy6a) [reply src/components/product/ProductCard.component.tsx:17]: done

File: src/App.tsx (hunk 3 above)
Comment (3apy6a) [reply src/App.tsx:12]: done

File: src/App.tsx (hunk 4 above)
Comment (3apy6a) [reply src/App.tsx:23]: done

**Review by vadym-zinchenko-moc — APPROVED** [review]

**Pull request merged**